### PR TITLE
[Zookeeper] Add metric type mapping for mntr datastream

### PR DESCRIPTION
**Pull request by agithomas**
- Enhancement

## What does this PR do?

Added metric_type mapping for `mntr` datastream.

## Checklist

- [x] I have reviewed [tips for building integrations](https://github.com/elastic/integrations/blob/main/docs/tips_for_building_integrations.md) and this pull request is aligned with them.
- [x] I have verified that all data streams collect metrics or logs.
- [x] I have added an entry to my package's `changelog.yml` file.
- [x] I have verified that Kibana version constraints are current according to [guidelines](https://github.com/elastic/elastic-package/blob/master/docs/howto/stack_version_support.md#when-to-update-the-condition).

## Author's Checklist

- [ ]

## How to test this PR locally

- elastic-package build
- Mapping tests completed
- Dashboard tests completed

## Related issues

- https://github.com/elastic/integrations/issues/7752

### CODE REVIEW

**Comment by agithomas** [comment]
Validation in mappings.
```
"zookeeper": {
        "properties": {
          "mntr": {
            "properties": {
              "approximate_data_size": {
                "type": "long",
                "time_series_metric": "gauge"
              },
              "ephemerals_count": {
                "type": "long",
                "time_series_metric": "gauge"
              },
              "followers": {
                "type": "long",
                "time_series_metric": "gauge"
              },
              "hostname": {
                "type": "keyword",
                "ignore_above": 1024
              },
              "latency": {
                "properties": {
                  "avg": {
                    "type": "long",
                    "time_series_metric": "gauge"
                  },
                  "max": {
                    "type": "long",
                    "time_series_metric": "gauge"
                  },
                  "min": {
                    "type": "long",
                    "time_series_metric": "gauge"
                  }
                }
              },
              "max_file_descriptor_count": {
                "type": "long",
                "time_series_metric": "gauge"
              },
              "num_alive_connections": {
                "type": "long",
                "time_series_metric": "gauge"
              },
              "open_file_descriptor_count": {
                "type": "long",
                "time_series_metric": "gauge"
              },
              "outstanding_requests": {
                "type": "long",
                "time_series_metric": "gauge"
              },
              "packets": {
                "properties": {
                  "received": {
                    "type": "long",
                    "time_series_metric": "gauge"
                  },
                  "sent": {
                    "type": "long",
                    "time_series_metric": "gauge"
                  }
                }
              },
              "pending_syncs": {
                "type": "long",
                "time_series_metric": "gauge"
              },
              "server_state": {
                "type": "keyword",
                "ignore_above": 1024
              },
              "synced_followers": {
                "type": "long",
                "time_series_metric": "gauge"
              },
              "watch_count": {
                "type": "long",
                "time_series_metric": "gauge"
              },
              "znode_count": {
                "type": "long",
                "time_series_metric": "gauge"
              }
            }
```

File: packages/zookeeper/data_stream/mntr/fields/fields.yml [hunk 1]
```diff
@@ -7,54 +7,67 @@
...
       description: |
         Number of file descriptors open by the ZooKeeper process.
     - name: outstanding_requests
       type: long
+      metric_type: gauge
       description: |
         Number of outstanding requests that need to be processed by the cluster.
     - name: packets.received
       type: long
+      metric_type: gauge
```
Comment (lalit-satapathy) [inline packages/zookeeper/data_stream/mntr/fields/fields.yml:60]: Is this a different field compared to the connection data stream, there it is marked counter?

File: packages/zookeeper/data_stream/mntr/fields/fields.yml [hunk 2]
```diff
@@ -7,54 +7,67 @@
...
       description: |
         Number of ZooKeeper network packets received.
     - name: packets.sent
```
Comment (lalit-satapathy) [inline packages/zookeeper/data_stream/mntr/fields/fields.yml:63]: Is this a different field compared to the connection data stream, there it is marked counter?

File: packages/zookeeper/data_stream/mntr/fields/fields.yml [hunk 3]
```diff
@@ -7,54 +7,67 @@
...
       description: |
         Number of ephemeral znodes.
     - name: followers
       type: long
+      metric_type: gauge
       description: |
         Number of followers seen by the current host.
     - name: max_file_descriptor_count
       type: long
+      metric_type: gauge
```
Comment (lalit-satapathy) [inline packages/zookeeper/data_stream/mntr/fields/fields.yml:30]: Is this value always fixed or sometime changes?

**Review by lalit-satapathy — COMMENTED** [review]
@agithomas, is there a reference to correlate the metrics types captured here?

File: packages/zookeeper/data_stream/mntr/fields/fields.yml (hunk 2 above)
Comment (agithomas) [reply packages/zookeeper/data_stream/mntr/fields/fields.yml:63]: In connections datastream, the data monotonically increase for a connected client. Data captured at various timeseries is provided below

```
agikthomas@service-integration-dev-idc-1:~$ echo cons | nc localhost 21811
 /172.27.0.1:61874[0](queued=0,recved=1,sent=0)
 /172.27.0.1:64796[1](queued=0,recved=24,sent=24,sid=0x100578e60f20006,lop=PING,est=1694441976312,to=30000,lcxid=0x0,lzxid=0x20000001d,lresp=1851800129,llat=0,minlat=0,avglat=0,maxlat=4)
 /172.27.0.1:63332[1](queued=0,recved=22,sent=22,sid=0x100578e60f20007,lop=PING,est=1694441995457,to=30000,lcxid=0x0,lzxid=0x20000001d,lresp=1851799260,llat=1,minlat=0,avglat=0,maxlat=3)

agikthomas@service-integration-dev-idc-1:~$ echo cons | nc localhost 21811
 /172.27.0.1:64796[1](queued=0,recved=25,sent=25,sid=0x100578e60f20006,lop=PING,est=1694441976312,to=30000,lcxid=0x0,lzxid=0x20000001d,lresp=1851810130,llat=1,minlat=0,avglat=0,maxlat=4)
 /172.27.0.1:65292[0](queued=0,recved=1,sent=0)
 /172.27.0.1:63332[1](queued=0,recved=23,sent=23,sid=0x100578e60f20007,lop=PING,est=1694441995457,to=30000,lcxid=0x0,lzxid=0x20000001d,lresp=1851809268,llat=0,minlat=0,avglat=0,maxlat=3)

agikthomas@service-integration-dev-idc-1:~$ echo cons | nc localhost 21811
 /172.27.0.1:64796[1](queued=0,recved=26,sent=26,sid=0x100578e60f20006,lop=PING,est=1694441976312,to=30000,lcxid=0x0,lzxid=0x20000001d,lresp=1851820140,llat=0,minlat=0,avglat=0,maxlat=4)
 /172.27.0.1:63494[0](queued=0,recved=1,sent=0)
 /172.27.0.1:63332[1](queued=0,recved=24,sent=24,sid=0x100578e60f20007,lop=PING,est=1694441995457,to=30000,lcxid=0x0,lzxid=0x20000001d,lresp=1851819271,llat=0,minlat=0,avglat=0,maxlat=3)

```

File: packages/zookeeper/data_stream/mntr/fields/fields.yml (hunk 3 above)
Comment (agithomas) [reply packages/zookeeper/data_stream/mntr/fields/fields.yml:30]: The definition goes as `Maximum number of file descriptors that a zookeeper server can open.`. Until modified to a higher or lower number, this value remain constant.

**Comment by agithomas** [comment]
> @agithomas, is there a reference to correlate the metrics types captured here?

No official reference document is available.

File: packages/zookeeper/data_stream/mntr/fields/fields.yml (hunk 3 above)
Comment (lalit-satapathy) [reply packages/zookeeper/data_stream/mntr/fields/fields.yml:30]: In this case, should it be a metric?

Comment (agithomas) [reply packages/zookeeper/data_stream/mntr/fields/fields.yml:30]: It is not incorrect to assign a `gauge` metric_type . Also, we follow the same approach in other packages including system where a max value is assigned a `gauge` metric_type

**Review by ritalwar — APPROVED** [review]
LGTM!

**Review by lalit-satapathy — APPROVED** [review]
LGTM

**Pull request merged**